AbstractFactoryPatternExample/src/main/cpp/src/abstract_dessert_factory.cpp

```diff
@@ -0,0 +1,29 @@
+#include "abstract_dessert.h"
+#include "abstract_dessert_factory.h"
+#include "cookie_factory.h"
+#include "cheesecake_factory.h"
+
+/* Implementation of the Abstract Dessert Factory */
+
+/* Return a pointer of type AbstractDessertFctory so each concrete factory
+ * can return an appropriate object. Notice that the pointers being returned are
+ * declared in their respective header files; they are implementations of the
+ * AbstractDessertFactory, but by virtue of inheritance, they are also of type
+ * AbstractDessertFactory. */
+AbstractDessertFactory* AbstractDessertFactory::getDessertFactory(int type){
+    switch(type){
+        case 1:
+            return new CookieFactory;
+            break;
+        case 2:
+            return new CheesecakeFactory;
+            break;
+        //default:
+        //    throw new std::exception("Invalid dessert selection!");
+    }
+}
```

Comment:
It could have a case where there is no default return value, where either an exception needs to be thrown or a null needs to be returned.